test/index.html

```diff
@@ -0,0 +1,250 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>QUnit tests for impress.js</title>
+  <link rel="stylesheet" href="qunit/qunit.css">
+</head>
+
+
+
+<body class="impress-not-supported">
+  <!-- HTML elements to be tested with QUnit should typically be inside
+       a qunit-fixture div. Unfortunately that just won't work with impress.js
+       as it wants the div#impress element to be a direct child of body. Or
+       something, I don't know, but it didn't work.
+       
+       So we just have an empty qunit-fixture, and then our impress steps
+       just below it. This just means you have to make sure that each
+       QUnit.test() cleans up after itself, can't rely on QUnit to do it now.
+  -->
+  <div id="qunit-fixture">
+  </div>
+
+  <div id="impress">
+    <div class="step" data-x="-1000" data-y="0">First slide</div>
+    <div class="step" data-x="-800" data-y="0">Second slide</div>
+    <div class="step" data-x="-600" data-y="0">Third slide</div>
+    <div class="step" id="fourth" data-x="-400" data-y="0">Fourth slide</div>
+  </div>
+
+  <div id="qunit"></div>
+
+  <script src="../js/impress.js"></script>
+  <script src="qunit/qunit.js"></script>
+  <script src="helpers.js"></script>
+  <script>
+    // In case tests didn't complete, we are left with a hash/anchor pointing somewhere. But we want to start from scratch:
+    window.location.hash = "";    
+    // I don't understand exactly why, but we must call init() before the actual tests
+    impress().init();
+    
+    QUnit.test( "Initialize Impress.js", function( assert ) {
+      assert.ok( impress, 
+                 "impress declared in global scope" );
+// This is the one that doesn't work. Test 17 fails if this is done inside a test()
+//      assert.deepEqual( impress().init(), undefined,
+//                        "impress().init() called." );
```

Comment:
I believe it's worth tracking down the root cause of everything and keep new technical debt at a minimum. Since you are the one currently inspecting the codebase, it is much easier for you to do this.

The reason why I suggest tracking and exposing the root cause of all problems via PR comments is that it allows us to understand better the current state of the code in order to think about better strategies to tackle those problems now or in the future.